### ISSUE TITLE
`npm unlink` won't remove global links because gentlyRm takes its job very seriously

### ISSUE THREAD

**Issue by othiym23** [npm/npm#9980]
Only affects `npm@3`, with the following output:

```
npm ERR! gentlyRm containing path /usr/local/lib/node_modules/npm isn't under npm's control
npm WARN install:npm Refusing to delete: /usr/local/bin/npm not in /usr/local/lib/node_modules/npm
npm verb install:npm Error: Refusing to delete: /usr/local/bin/npm not in /usr/local/lib/node_modules/npm
```

Repro case is to run `make link` from a checkout of the npm repository, and then run `npm unlink` (with no arguments) afterwards.

**Comment by denysdovhan** [comment]
I have the same problem.

**Comment by jasonkarns** [comment]
dupe of #9680 ?

**Comment by denysdovhan** [comment]
Looks like it is the same

**Comment by iarna** [comment]
I didn't get time to get this in for 3.3.10, so I'm pushing this to next week.

**Comment by denysdovhan** [comment]
No problem. I'm just still use Unix `rm` for unlinking.

**Comment by iarna** [comment]
This was pushed out One More Week, because while I have a PR available, we would like to take the opportunity to write some unit tests for it, now that it's been rewritten to be unit-testable.

**Comment by denysdovhan** [comment]
Okay, not problem.

**Comment by vproman** [comment]
v3.4.1 fixes this issue (per #10270) if `npm install -g npm` doesn't get you 3.4.1+ because it's not yet in the repo, do `npm install -g npm@v3.4.1`.
